### ISSUE TITLE
Non-beautified JSON output

### ISSUE THREAD

**Issue by thannaske** [jandelgado/rabtap#31]
For automation & further processing of messages it's somewhat annoying that the JSON-formatted output is beautified. Therefore you can't parse `rabtap`'s output line by line. You need to assemble the whole output line by line until the JSON message is completed.

It would be awesome if there would be a flag to disable beautification of the JSON-output (or to opt-in to it).

**Comment by jandelgado** [comment]
Since the problem seems to be your JSON parser having problems handling a JSON stream, I would not try to fix it in rabtap. An easy solution might be to pipe the output through a JSON formatter, e.g.
```
$ rabtap tap .... --json | json_reformat -ms | my-consumer
```
Wich produces one JSON document per line.

See http://manpages.ubuntu.com/manpages/xenial/man1/json_reformat.1.html

**Comment by thannaske** [comment]
Unfortunately `json_reformat -ms` doesn't produce output for me in any way. I already tried to use `jq`, but this just worked at the first message - further ones could not be parsed.

Of course you are right - it's a problem in my pipeline, but it would be way easier for anyone who tries to use rabtap within a processing toolchain to choose a one-liner JSON output. They could focus on building their pipeline using your awesome tool instead of figuring out how to get multi-line JSON output parsed properly ;-)

**Comment by jandelgado** [comment]
It seem that `json_reformat` does some buffering, delaying data being receiver by `tee`.

I'll implement a `--no-pretty-print` feature in the next days, in the meantime, you could use this go tool, which works as expected:

```golang
// json_flat: read json stream from stdin until EOF and reformat each document
// to fit on a single line. Example usage:
//   rabtap tap amq.topic:# --json | json_flat | tee messages.log
// Jan Delgado 11-2019
package main

import (
	"encoding/json"
	"fmt"
	"io"
	"log"
	"os"
)

func main() {
	d := json.NewDecoder(os.Stdin)
	var m map[string]interface{}
	for {
		err := d.Decode(&m)
		if err == io.EOF {
			return
		}
		if err != nil {
			log.Fatal(err)
		}

		json, err := json.Marshal(m)
		if err != nil {
			log.Fatal(err)
		}
		fmt.Println(string(json))
		os.Stdout.Sync()
	}
}
```

**Comment by thannaske** [comment]
Great to hear! Thanks, I really appreciate it.

**Comment by jandelgado** [comment]
Implemented in version 1.21: use new `--format=json-nopp` option
